lib/Xud.ts

```diff
@@ -76,7 +77,10 @@ class Xud {
       if (!this.raidenClient.isDisabled()) {
         initPromises.push(this.raidenClient.init());
       }
-      this.pool = new Pool(this.config.p2p, loggers.p2p, this.db, this.lndbtcClient, this.lndltcClient);
+
+      const swaps = new Swaps(loggers.swaps);
+
+      this.pool = new Pool(this.config.p2p, loggers.p2p, this.db, swaps);
 
       this.orderBook = new OrderBook(this.logger, this.db.models, this.pool, this.lndbtcClient, this.raidenClient);
```

Comment:
what about having `swaps` to handle the lnd/raiden clients, and to be used also for creating invoices, expose the swaps events, pubkeys, etc. ?

Reply:
Yeah I like that, makes sense.